src/App.tsx

```diff
@@ -1,61 +1,137 @@
+import { FC, useState } from 'react';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const findUser = (param: number) => {
+  return usersFromServer
+    .find(user => user.id === param)
+    || null;
+};
+
+const preparedTodos = todosFromServer.map(todo => {
+  const user = findUser(todo.userId);
+
+  return {
+    ...todo,
+    user,
+  };
+});
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState(preparedTodos);
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [currentUserId, setCurrentUserId] = useState(0);
+  const [hasCurrentUserIdError, setHasCurrentUserIdError] = useState(false);
+
+  const maxId = todos.map((todo) => todo.id).sort((a, b) => b - a)[0];
+
+  const handleUserSelectChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setCurrentUserId(+e.target.value);
+    setHasCurrentUserIdError(false);
+  };
+
+  const handleOnInput = ((e: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(e.target.value);
+    setHasTitleError(false);
+  });
+
+  const handleOnSubmit = ((e: React.ChangeEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!title && !currentUserId) {
+      setHasTitleError(!hasTitleError);
+      setHasCurrentUserIdError(!hasCurrentUserIdError);
+
+      return;
+    }
+
+    if (!title) {
+      setHasTitleError(!hasTitleError);
+
+      return;
+    }
+
+    if (!currentUserId) {
+      setHasCurrentUserIdError(!hasCurrentUserIdError);
+
+      return;
+    }
+
+    const newPost = {
+      id: maxId + 1,
```

Comment:
You can calculate a new `id` on the same time when you're creating new object, according to the checklist. 
```suggestion
      id:  (todos.map((todo) => todo.id).sort((a, b) => b - a)[0] + 1);
```